cloudformation/external-slurmdbd/external_slurmdbd/external_slurmdbd_stack.py

```diff
@@ -0,0 +1,274 @@
+import json
+
+from aws_cdk import App, CfnParameter, Fn, Stack
+from aws_cdk import aws_autoscaling as autoscaling
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_elasticloadbalancingv2 as elbv2
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_logs as logs
+from constructs import Construct
+
+from pcluster.constants import EXTERNAL_SLURMDBD_ASG_SIZE
+
+
+class ExternalSlurmdbdStack(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+        self.stack = Stack(scope=scope, id=construct_id, **kwargs)
+
+        # define networking stuff
+        self.vpc_id = CfnParameter(
+            self, "VPC_id", type="String", description="The VPC to be used for the Slurmdbd stack."
+        )
+        self.vpc = ec2.Vpc.from_lookup(self, "VPC", vpc_id=self.vpc_id.value_as_string)
+
+        self.subnet_id = CfnParameter(
+            self, "SubnetId", type="String", description="The Subnet to be used for the Slurmdbd stack."
+        )
+        self.subnet = ec2.Subnet.from_subnet_id(self, "subnet", subnet_id=self.subnet_id.value_as_string)
+
+        # define Target Group
+        self._external_slurmdbd_target_group = self._add_external_slurmdbd_target_group()
+
+        # define Network Load Balancer (NLB)
+        self._external_slurmdbd_nlb = self._add_external_slurmdbd_load_balancer(
+            target_group=self._external_slurmdbd_target_group
+        )
+
+        # Define additional CloudFormation parameters for dna.json to pass to cookbook
+        self.dbms_uri = CfnParameter(self, "DBMSUri", type="String", description="DBMS URI for Slurmdbd.")
+        self.dbms_username = CfnParameter(
+            self, "DBMSUsername", type="String", description="DBMS Username for Slurmdbd."
+        )
+        self.dbms_password_secret_arn = CfnParameter(
+            self, "DBMSPasswordSecretArn", type="String", description="Secret ARN for DBMS password."
+        )
+        self.dbms_database_name_para = CfnParameter(
+            self, "DBMSDatabaseName", type="String", description="DBMS Database Name for Slurmdbd."
+        )
+        self.munge_key_secret_arn = CfnParameter(
+            self, "MungeKeySecretArn", type="String", description="Secret ARN for Munge key."
+        )
+
+        # use cfn-init and cfn-hup configure instance
+        self._cfn_init_config = self._add_cfn_init_config()
+
+        # create management security group with SSH access from anywhere (TEMPORARY!)
+        self._ssh_server_sg, self._ssh_client_sg = self._add_management_security_groups()
```

Comment:
Please add another pair of security groups for the slurm accounting traffic across between cluster head node and external slurmdbd instance via port 6819.

Reply:
Done!